pkg/tfbridge/provider.go

```diff
@@ -168,6 +183,34 @@ func NewProvider(ctx context.Context, host *provider.HostClient, module string,
 	return p
 }
 
+func newMuxWithProvider(ctx context.Context, host *provider.HostClient,
+	module, version string, info ProviderInfo, pulumiSchema []byte,
+) (pulumirpc.ResourceProviderServer, error) {
+	var mapping muxer.DispatchTable
+	if m, found, err := metadata.Get[muxer.DispatchTable](info.GetMetadata(), "mux"); err != nil {
+		return nil, err
+	} else if found {
+		mapping = m
+	} else {
+		return nil, fmt.Errorf("missing pre-computed muxer mapping")
+	}
+
+	servers := []muxer.Endpoint{{
+		Server: func(host *provider.HostClient) (pulumirpc.ResourceProviderServer, error) {
+			return newProvider(context.Background(), host, module, version, info.P, info, pulumiSchema), nil
```

Comment:
Document why context.Background() instead of `ctx` from the parameter?

Reply:
@t0yv0 I merged a PR of @iwahbe -> https://github.com/tmeckel/pulumi-terraform-bridge/pull/1 I think that's an oversight of him, but perhaps there's a reason for this.

Reply:
I think before the refactor the outer code didn't have the context parameter. I think it does not matter *yet* but it's best to propagate `ctx` rather than initiate a new context chain - that should always be unusual and documented with a comment. Things like implicit loggers are passed via context sometimes in this codebase and this can get interesting.

Reply:
@t0yv0 I'd prefer to propagate the ctx down as well. As I said I think it's a copy-paste fault.

Reply:
@iwahbe @t0yv0 propagating down `ctx` now